banzai_nres/frames.py

```diff
@@ -62,26 +63,69 @@ def table(self):
         self.classification = None
-        self._hdu_names = [hdu.name for hdu in hdu_list]
-
-    def __getitem__(self, item):
-        if item not in self._hdu_names and not isinstance(item, int):
-            raise ValueError('Requested HDU name is not in frame')
-        if isinstance(item, int):
-            return self._hdus[item]
-        return self._hdus[self._hdu_names.index(item)]
 
-    def __contains__(self, item):
-        return item in self._hdu_names
+        self._traces = None
+        self._profile = None
+        self._blaze = None
+        self._weights = None
+        self._fibers = None
+        self._features = None
+        self._wavelengths = None
+        self._spectrum = None
+        self._ccf = None
+        self.summary_figures = []
+
+    def get_output_data_products(self, runtime_context):
+        if self.obstype != 'TARGET':
+            return super().get_output_data_products(runtime_context)
+        else:
+            filename_1d = self.get_output_filename(runtime_context)
+            filename_1d = filename_1d.replace('.fits', '-1d.fits')
+            filename_2d = filename_1d.replace('-1d.fits', '-2d.fits')
+            filename_summary = filename_1d.replace('-1d.fits', '-summary.pdf').replace('.fz', '')
+
+            frame_1d_meta = self.meta.copy()
+            # Remove any name from the header of the primary hdu
+            if 'EXTNAME' in frame_1d_meta:
+                frame_1d_meta.pop('EXTNAME')
+            frame_1d = LCOObservationFrame([HeaderOnly(frame_1d_meta), self['SPECTRUM1D'], self['CCF']],
+                                           os.path.join(self.get_output_directory(runtime_context), filename_1d))
+            fits_1d = frame_1d.to_fits(runtime_context)
+            fits_1d['SPECTRUM1D'].name = 'SPECTRUM'
+            fits_1d[0].header['L1ID2D'] = filename_2d
+            output_product_1d = DataProduct.from_fits(fits_1d, filename_1d, self.get_output_directory(runtime_context))
+
+            frame_2d = LCOObservationFrame([hdu for hdu in self._hdus if hdu.name not in ['SPECTRUM1D', 'CCF']],
+                                           os.path.join(self.get_output_directory(runtime_context), filename_2d))
+            fits_2d = frame_2d.to_fits(runtime_context)
+            fits_2d[0].header['L1ID1D'] = filename_1d
+            output_product_2d = DataProduct.from_fits(fits_2d, filename_2d, self.get_output_directory(runtime_context))
+
+            summary_pdf_meta = fits_utils.sanitize_header(self.meta.copy())
+            for keyword in ['NAXIS1', 'NAXIS2', 'EXTNAME', 'EXTVER']:
+                if keyword in summary_pdf_meta:
+                    summary_pdf_meta.pop(keyword)
+
+            summary_buffer = BytesIO()
+            pp = PdfPages(summary_buffer, keep_empty=False, metadata=dict(summary_pdf_meta))
+            # make sure we pop all the fits garbage before we write the meta data.
```

Comment:
this comment seems out of place? is that supposed to go above the keyword popping logic you have above?

Reply:
Fixed.